src/backend/cdb/cdbmutate.c

```diff
@@ -1098,7 +1158,7 @@ add_slice_to_motion(Motion *motion,
 				/* broadcast */
 				motion->plan.flow = makeFlow(FLOW_REPLICATED, numsegments);
 				motion->plan.flow->locustype = CdbLocusType_Replicated;
-
+				motion->plan.plan_rows *= numsegments;
```

Comment:
Did you check that this code is applied only for motions created during apply_motion?